### PR TITLE
Implement API key storage via keyring

### PR DESCRIPTION
**Pull request by dcolli23**
- Implements a really simple API key storage solution so that no keys are committed on accident. This uses python bindings to GNOME keyring which is a pretty standard method of storing secrets.
- Opted to use keyring instead of environment or configuration files as keyring supports encryption.
- Due to this, we could also store robot credentials using keyring, preventing us from having to specify usernames/passwords or hard-code them. This is on my TODO list.

### CODE REVIEW

**Comment by dcolli23** [comment]
@Saketspradhan what do you think of this for storing API keys?

File: src/conq/perception_lib/sam_inference.py [hunk 1]
```diff
@@ -17,7 +12,10 @@
...
 import replicate
 
-os.environ['REPLICATE_API_TOKEN'] = '***ADD API TOKEN HERE***'
+from conq.api_keys import REPLICATE_API_KEY
+
+os.environ['REPLICATE_API_TOKEN'] = REPLICATE_API_KEY.get()
+
 
 def run_sam_inference(image_url):
     start_time = time.time()
```
Comment (dcolli23) [inline src/conq/perception_lib/sam_inference.py:21]: @Saketspradhan two lines below this one (so, line 23), it looks like there's an API key hard-coded. Is that one?

Comment (Saketspradhan) [reply src/conq/perception_lib/sam_inference.py:21]: Which file/line are you referring to? Is it something to do with the Roboflow API?

**Comment by Saketspradhan** [comment]
Yeah, this is a really good way to manage any keys that we would be using, so as to mitigate any future mishaps from happening

File: src/conq/perception_lib/sam_inference.py (hunk 1 above)
Comment (Saketspradhan) [reply src/conq/perception_lib/sam_inference.py:21]:
```
def run_sam_inference(image_url):
    start_time = time.time()
    image_url = replicate.run(
        "pablodawson/segment-anything-automatic:14fbb04535964b3d0c7fad03bb4ed272130f15b956cbedb7b2f20b5b8a2dbaa0",
        input={
            "image": image_url
        }
    )
```

Are you referring to the `hash-key` in this block?

Comment (Saketspradhan) [reply src/conq/perception_lib/sam_inference.py:21]: That's not a problem, it is public access within replicate itself

**Review by Saketspradhan — APPROVED** [review]
Thanks for taking the time to do this @dcolli23! Really appreciate it!

**Comment by Saketspradhan** [comment]
@dcolli23 Do you wanna merge and close it, or should I do it?

File: src/conq/perception_lib/sam_inference.py (hunk 1 above)
Comment (dcolli23) [reply src/conq/perception_lib/sam_inference.py:21]: Yep, that's what I was referring to. Sounds like it's fine to have it in here so I'll resolve this conversation.

**Comment by dcolli23** [comment]
> Thanks for taking the time to do this @dcolli23! Really appreciate it!

No problem! Happy that it'll be useful. I'll go ahead and squash and merge

**Pull request merged**